### ISSUE TITLE
Date picker is invisible to screen readers when starting/finishing a book

### ISSUE THREAD

**Issue by Flameborn** [bookwyrm-social/bookwyrm#1044]
After pressing *Start Reading* or *Finish Reading* on a book's page, a dialog pops up with the current date. This date picker is invisible to screen readers, nor can be focused/edited via the keyboard.

This might be due to an *ARIA-hidden* flag in the template.

**Comment by mouse-reeve** [comment]
I'm having trouble replicating this, I'm getting the same (janky) behavior with the date picker wherever it's showing up -- there aren't any aria-hidden attributes at play as far as I know. This may be yet another version of the evident poor support for the html date field?

**Comment by Flameborn** [comment]
Ah, in this case I think the date picker not being accessible enough is definitely a Safari/VoiceOver issue. Unfortunately each browser has their own implementation, some of them are better than others.

Interestingly, the date picker is reachable when editing a book, although VoiceOver does not seem to read the new values when the date is modified.

When starting/finishing a book, however, the date picker is not visible at all, either via page navigation via the VoiceOver cursor, or via the tab key.

Do you think one of these implementations would work? https://www.digitala11y.com/accessible-datepickers-roundup

**Comment by mouse-reeve** [comment]
Ah that's a great list, I'll go through them and see what looks promising. Thank you!

**Comment by mouse-reeve** [comment]
@Flameborn I'm sorry I haven't made headway on this one - it's really important to me that this, a major part of the application actually _works_, but I'm not sure how to fix it. I'm wondering if a holdover solution would be to include an alternative date field for screen readers only, that uses a standard text input and prompts the user to enter the date as `YYYY-mm-dd`? Generally I really want to avoid having separate markup for screen readers, so that it doesn't double the work of maintaining the app and increase the likelihood of accessible version falling behind, but I probably won't get a chance to properly fix this in a timeline that I would feel good about.

**Comment by Flameborn** [comment]
No worries. ☺️
Part of me just wants to say that you should not worry about this, as it's a bad implementation via browsers, but of course that does not make it usable via a screen reader.

I think the two possible solutions would be to either put this off until browsers have a better implementation of the date picker, which is not guaranteed, or to replace it with a more accessible one, but I'm not sure which implementation would work in this case.

**Comment by joachimesque** [comment]
In the modal form, the `input type=date` are placed inside the `label`. It is valid HTML, but it might be the cause of the wrong behaviour on this form and not on the Author form. I'm proposing a fix in #1154, some testing will be necessary after the PR is accepted and deployed.
